src/Service/Services/ExecutionHelper.cs

```diff
@@ -208,24 +227,33 @@ private static object PreParseLeaf(IMiddlewareContext context, string leafJson)
 
         public static bool RepresentsNullValue(JsonElement element)
         {
+            // TODO: why does this not check the element kind?
             return string.IsNullOrEmpty(element.ToString()) && element.GetRawText() == "null";
         }
 
-        protected static bool TryGetPropertyFromParent(IMiddlewareContext context, out JsonElement jsonElement)
+        protected static bool TryGetPropertyFromParent(
+            IPureResolverContext context,
+            out JsonElement jsonElement)
         {
-            JsonDocument result = JsonDocument.Parse(JsonSerializer.Serialize(context.Parent<JsonElement>()));
+            // TODO : why is this done?
+            JsonDocument result = 
```

Comment:
Hey @Aniruddh25 ,

why is this done in the resolver class. This is rather inefficient. If you think about objects in list or objects in nested lists this is really not nice.

Is the reason that you can locally dispose? Although I see no dispose here.

If you need to dispose objects after serialization Hot Chocolate allows to attach those handles to the result and clean them up in the transport after everything has been written to the output.

Why are we not using the parent here bit are copying the object in this way?